Gemfile

```diff
@@ -6,7 +6,7 @@ gem 'net-dns'
 gem 'bit-struct'
 gem 'geoip-c'
 gem 'maxmind-db', '~> 1.0.0'
-gem 'recog', '>= 2.3.21'
+gem 'recog', '~> 3.0'
```

Comment:
This is probably fine, just highlighting the dropping of the patch version. The Gemfile.lock specifies recog 3.0.2 (which this covers)

Reply:
The `recog` dependency is specified to be a version that is `>= 3.0` but `< 3.1` so that it accepts all 3.0 patch versions. Running `bundle install` fins gems that meet the specified requirements, in this case recog `3.0.2` is fetched and then writes a snapshot of all of the gems and versions installed to the `Gemfile.lock` file.